src/map/options.ts

```diff
@@ -420,4 +420,18 @@ export class MapOptions extends OptionsGroup {
     public colorScale(): Plotly.ColorScale {
         return COLOR_MAPS[this.palette.value];
     }
+
+    public logLinearLabelSwitch = (axis: AxisOptions, axisName: string) => {
```

Comment:
Why a member variable with type function instead of a member function? This is not supposed to be changed directly by code using it, right?

```suggestion
    public logLinearLabelSwitch(axis: AxisOptions, axisName: string): void {
```

----

Also, this needs documentation and maybe a better name? What do you think of `setLogLabel` (using action for the function name) and document that this changes between log/linear scale label as appropriate?